src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,146 @@
+        RecommendationRequest recommendationRequest = 
+        recommendationRequestRepository
+            .findById(id)
+            .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequestRepository.delete(recommendationRequest);
+
+        return genericMessage("RecommendationRequest with id %s deleted".formatted(id));
+    }
+
+    /**
+     * The user who posted a RecommendationRequest can delete their RecommendationRequest
+     * 
+     * @param id the id of the RecommendationRequest to delete
+     * @return a message indicating that the RecommendationRequest was deleted
+     */
+    @Operation(summary = "User can delete their RecommendationRequest")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @DeleteMapping("")
+    public Object deleteRecommendationRequestAsUser(@Parameter(name = "id") @RequestParam Long id) {
+        User currentUser = getCurrentUser().getUser(); 
+        RecommendationRequest recommendationRequest = 
+        recommendationRequestRepository
+            .findByIdAndUser(id, currentUser)
+            .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequestRepository.delete(recommendationRequest);
+
+        return genericMessage("RecommendationRequest with id %s deleted".formatted(id));
+    }
+
+    /**
+     * Any admin can update a single Recommendation Request
+     * 
+     * @param id       the id of the Recommendation Request to update
+     * @param incoming the updated Recommendation Request
+     * @return the updated Recommendation Request object
+     */
+    @Operation(summary = "An admin can update a single recommendation request")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("/admin")
+    public RecommendationRequest updateRecommendationRequestAsAdmin(
+        @Parameter(name = "id") @RequestParam Long id,
+        @RequestBody @Valid RecommendationRequest incoming) {
+
+        RecommendationRequest recommendationRequest = 
+            recommendationRequestRepository
+                .findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequest.setProfessorName(incoming.getProfessorName());
+        recommendationRequest.setProfessorEmail(incoming.getProfessorEmail());
+        recommendationRequest.setRequesterName(incoming.getRequesterName());
+        recommendationRequest.setRecommendationTypes(incoming.getRecommendationTypes());
+        recommendationRequest.setDetails(incoming.getDetails());
+        recommendationRequest.setStatus(incoming.getStatus());
+        recommendationRequest.setSubmissionDate(incoming.getSubmissionDate());
+        recommendationRequest.setCompletionDate(incoming.getCompletionDate());
+
+        recommendationRequestRepository.save(recommendationRequest);
+
+        return recommendationRequest;
+    }
```

Comment:
I'm not sure this is a good idea.   What's the use case for it?  Meaning: tell me a story about a student, a professor and an admin, where the admin ends up needing to do this for a good reason.

The only stories I can come up with involve some kind of "forging of records", so that seems.. bad?